System/Console/Docopt/Public.hs

```diff
@@ -47,32 +48,46 @@ import Data.Maybe (fromMaybe)
 import System.Console.Docopt.Types
 import System.Console.Docopt.ApplicativeParsec (ParseError)
 import System.Console.Docopt.OptParse
-
+import System.IO
 
 -- | Parse command line arguments.
 parseArgs :: Docopt -> [String] -> Either ParseError Arguments
 parseArgs parser = getArguments (optFormat parser)
 
--- | Same as 'parseArgs', but 'exitWithUsage' on parse failure. E.g.
+-- | Same as 'parseArgs', but 'dieWithUsage' on parse failure. E.g.
 --
 -- > args <- parseArgsOrExit patterns =<< getArgs
-parseArgsOrExit :: Docopt -> [String] -> IO Arguments
-parseArgsOrExit parser argv = either (const $ exitWithUsage parser) return $ parseArgs parser argv
+parseArgsOrDie :: Docopt -> [String] -> IO Arguments
+parseArgsOrDie parser argv = either (const $ dieWithUsage parser) return $ parseArgs parser argv
 
--- | Exit after printing usage text.
+-- | Exit after printing usage text to stdout.
 exitWithUsage :: Docopt -> IO a
 exitWithUsage doc = do
   putStr $ usage doc
-  exitFailure
+  exitSuccess
+
+-- | Exit after printing usage text to stderr.
+dieWithUsage :: Docopt -> IO a
+dieWithUsage doc = die $ usage doc
```

Comment:
I didn't notice before, but I'm pretty amused by the fact that `dieWithUsage doc` is now a _longer_ (in char count) invocation than would be `die $ usage doc`. This is a "composition FTW" if I've ever seen one.